### DISCUSSION TITLE
helm-controller: detect drift in cluster

### DISCUSSION THREAD

**Discussion by jasaltvik** [fluxcd/flux2#155] [Proposals]
Sometimes a resource managed by the Helm Operator (through a `HelmRelease`) is deleted by mistake (e.g. with `kubectl`). It would be nice if the Helm Operator could detect this, and reconcile the cluster state with the contents of the Helm release at the next sync interval.

Slack discussion: https://cloud-native.slack.com/archives/CLAJ40HV3/p1597310105111800

**Comment by hiddeco** [comment]
An issue related to this can be found here: https://github.com/fluxcd/helm-controller/issues/186, but without any new progress to share.
